packages/app/src/cli/models/app/identifiers.ts

```diff
@@ -45,19 +47,22 @@ interface UpdateAppIdentifiersOptions {
  * @returns An copy of the app with the environment updated to reflect the updated identifiers.
  */
 export async function updateAppIdentifiers(
-  {app, identifiers, command}: UpdateAppIdentifiersOptions,
+  {app, identifiers, command, developerPlatformClient}: UpdateAppIdentifiersOptions,
   systemEnvironment = process.env,
 ): Promise<AppInterface> {
-  await Promise.all(
-    app.realExtensions.map(async (extension) => {
-      if (extension.isUuidManaged() && (await fileExists(extension.configurationPath))) {
-        const tomlContents = await readFile(extension.configurationPath)
-        const extensionConfig = decodeToml(tomlContents)
-        if ('uid' in extensionConfig) return
-        await writeFile(extension.configurationPath, `uid = "${extension.uid}"\n${tomlContents}`)
-      }
-    }),
-  )
+  if (developerPlatformClient.supportsAtomicDeployments) {
+    await Promise.all(
+      app.realExtensions.map(async (extension) => {
+        if (extension.isUuidManaged() && (await fileExists(extension.configurationPath))) {
+          const tomlContents = await readFile(extension.configurationPath)
+          const extensionConfig = decodeToml(tomlContents)
+          if ('uid' in extensionConfig) return
+          const updatedTomlContents = tomlContents.replace(/(type\s*=\s*".*")/, `$1\nuid = "${extension.uid}"`)
```

Comment:
Seems worth ensuring that the key is `type` and not `something_type`

```suggestion
          const updatedTomlContents = tomlContents.replace(/^(\s*type\s*=\s*".*")/, `$1\nuid = "${extension.uid}"`)
```

Reply:
Thanks for the suggestion! I've followed a similar approach.